hantec_api_ecommerce/controllers/main.py

```diff
@@ -0,0 +1,915 @@
+        provided in the JSON request body.
+
+        URL parameter:
+            - invoice (account.move): The invoice model instance.
+
+        JSON request body:
+            - amount (float): The amount to be paid.
+            - journal_id (int): The ID of the payment journal.
+            - payment_method_id (int): The ID of the payment method.
+
+        JSON response:
+            - success (str): A message indicating that the payment has been successfully registered.
+
+        Returns:
+            dict: A dictionary with a success message.
+
+        """
+        data = request.jsonrequest
+        amount = data.get("amount")
+        journal_id = data.get("journal_id")
+        payment_method_id = data.get("payment_method_id")
+
+        register_payment_wizard = request.env["account.payment.register"].with_context(
+            active_model="account.move", active_ids=[invoice.id]
+        )
+
+        wizard = register_payment_wizard.create(
+            {
+                "amount": amount,
+                "journal_id": journal_id,
+                "l10n_mx_edi_payment_method_id": payment_method_id,
+                "communication": "Payment for invoice %s" % invoice.name,
+            }
+        )
+
+        # Register the payment
+        wizard.action_create_payments()
+
+        return {"success": "The payment has been successfully registered."}
+
+    @route(
+        '/get_shipping_info/<model("sale.order"):order>',
+        methods=["GET"],
+        type="json",
+        auth="user",
+    )
+    def get_shipping_info(self, order=False):
+        """
+        Retrieves shipping information for a given sale order.
+
+        This function gathers shipping information related to the specified sale order,
+        including delivery address details and shipping records.
+
+        URL parameter:
+            - order (sale.order): The sale order model instance.
+
+        JSON response:
+            - message (str): A message indicating that the shipping data has been successfully retrieved.
+            - shipping_data (list): A list of dictionaries containing shipping information.
+
+        Returns:
+            dict: A dictionary with a success message and the shipping data.
+
+        """
+        # Delivery address data
+        shipping_address = order.partner_shipping_id
+        address_data = {
+            "name": shipping_address.name,
+            "phone": shipping_address.phone,
+            "email": shipping_address.email,
+            "street": shipping_address.street,
+            "street2": shipping_address.street2,
+            "city": shipping_address.city,
+            "state": (
+                shipping_address.state_id.name if shipping_address.state_id.name else ""
+            ),
+            "zip": shipping_address.zip,
+            "country": (
+                shipping_address.country_id.name if shipping_address.country_id else ""
+            ),
+        }
+
+        # Get related shipping records
```

Comment:
what about to move this logic to the model picking or order itself?

you will need to createa dir 
models/sale_order.py

and inside the sale_order.py you create the get method with all this logic, this is in order to separated the controllers and the data models

Reply:
this